src/analysis/tuneParam/tuneParam.m

```diff
@@ -0,0 +1,158 @@
+%
+% INPUT:
+%         LPProblem:     MILP as COBRA model structure
+%         contFunctName: Parameters structure containing the name and value.
+%                        A set of routine parameters will be added by the solver
+%                        but won't be reported.
+%         timelimit:     default is 10000 second
+%         nrepeat:       number of row/column permutation of the original
+%                        problem, reports robust results.
+%                        sets the CPX_PARAM_TUNINGREPEAT parameter
+%                        High values of nrepeat would require consequent
+%                        memory and swap.
+%         printLevel:    0/1/2/3
+%
+% OUTPUT:
+%         optimParam: structure of optimal parameter values directly usable as
+%                     contFunctName argument in solveCobraLP function
+% .. Author: Marouen Ben Guebila 24/07/2017
+
+if ~changeCobraSolver('ibm_cplex')
+    fprintf('This function requires IBM ILOG CPLEX');
+end
+if exist('timelimit','var')
+    contFunctName.tune.timelimit = timelimit;
+end
+if exist('nrepeat','var')
+    contFunctName.tune.repeat = nrepeat;
+end
+if exist('printLevel','var')
+    contFunctName.tune.display = printLevel;
+end
+interface='ILOGcomplex';
+if ~exist('contFunctName','var')
+    cpxControl=[];
+else
+    if isstruct(contFunctName)
+        cpxControl=contFunctName;
+    else
+        if ~isempty(contFunctName)
+            %calls a user specified function to create a CPLEX control structure
+            %specific to the users problem. A TEMPLATE for one such function is
+            %CPLEXParamSet
+            cpxControl=eval(contFunctName);
+        else
+            cpxControl=[];
+        end
+    end
+end
+
+if ~isfield(LPProblem,'A')
+    if ~isfield(LPProblem,'S')
+            error('Equality constraint matrix must either be a field denoted A or S.')
+    end
+    LPProblem.A=LPProblem.S;
+end
+
+if ~isfield(LPProblem,'csense')
+    nMet=size(LPProblem.A);
+    if printLevel>0
+        fprintf('%s\n','Assuming equality constraints, i.e. S*v=b');
+    end
+    %assuming equality constraints
+    LPProblem.csense(1:nMet,1)='E';
+end
+
+if ~isfield(LPProblem,'osense')
+    %assuming maximisation
+    LPProblem.osense=-1;
+    if printLevel>0
+        fprintf('%s\n','Assuming maximisation of objective');
+    end
+end
+
+if size(LPProblem.A,2)~=length(LPProblem.c)
+    error('dimensions of A & c are inconsistent');
+end
+
+if size(LPProblem.A,2)~=length(LPProblem.lb) || size(LPProblem.A,2)~=length(LPProblem.ub)
+    error('dimensions of A & bounds are inconsistent');
+end
+
+%get data
+[c,x_L,x_U,b,csense,osense] = deal(LPProblem.c,LPProblem.lb,LPProblem.ub,...
+    LPProblem.b,LPProblem.csense,LPProblem.osense);
+%modify objective to correspond to osense
+c=full(c*osense);
+
+%cplex expects it dense
+b=full(b);
+
+
+%call cplex
+tic;
+
+%complex ibm ilog cplex interface
+if ~isempty(csense)
```

Comment:
I don't think we should accept an empty `csense` vector (i.e. an input which has the csense  field, but it is empty). So the else clause is unnecessary.